rlclientlib/logger/async_batcher.h

```diff
@@ -151,11 +158,43 @@ namespace reinforcement_learning { namespace logger {
       }
     }
 
-    RETURN_IF_FAIL(collection_serializer.finalize(status));
-
+    if (_events_counter_status == events_counter_status::ENABLE) {
+      unsigned int buffer_start_event_number;
```

Comment:
unsigned int buffer_start_event_number  = _buffer_end_event_number looks a bit cleaner (the same about original_count)